### PR TITLE
Make provision for B-engine sensors

### PR DESCRIPTION
**Pull request by amishatishpatel**
There is no logic behind the sensors, this PR simply creates them in preparation for the logic.
* Where I've left defaults out, it's because I'm still not sure what the format of the value is.

The corresponding katsdpcontroller PR is at
* https://github.com/ska-sa/katsdpcontroller/pull/714

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] (N/A) If dependencies are added/removed: update `setup.cfg` and `.pre-commit-config.yaml`
- [x] (N/A) If modules are added/removed: use `sphinx-apidoc -efo doc/ src/` to update files in `doc/`
- [x] Ensure copyright notices are present and up-to-date
- [x] (N/A) If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Closes NGC-447.

### CODE REVIEW

File: src/katgpucbf/xbgpu/engine.py [hunk 1]
```diff
@@ -377,10 +377,37 @@ def _populate_sensors(self) -> None:
...
+                    "phase <radians>, ...)",
+                    initial_status=aiokatcp.Sensor.Status.NOMINAL,
+                )
+            )
+            sensors.add(
+                aiokatcp.Sensor(
+                    float,
+                    f"{output.name}.quantiser-gain",
+                    "Non-complex post-summation quantiser gain applied to this beam",
+                    default=0.0,
```
Comment (james-smith-za) [inline src/katgpucbf/xbgpu/engine.py:395]: Default gain of 0 doesn't make sense.
See: https://github.com/ska-sa/katgpucbf/blob/3b6cf9039403f5479e442b1d71f244f9820d4e1f/src/katgpucbf/xbgpu/engine.py#L334

File: src/katgpucbf/xbgpu/engine.py [hunk 2]
```diff
@@ -377,10 +377,37 @@ def _populate_sensors(self) -> None:
...
+                    initial_status=aiokatcp.Sensor.Status.NOMINAL,
+                )
+            )
+            sensors.add(
+                aiokatcp.Sensor(
+                    str,
+                    f"{output.name}.weight",
+                    "The summing weights applied to all the inputs of this beam",
+                    initial_status=aiokatcp.Sensor.Status.NOMINAL,
```
Comment (james-smith-za) [inline src/katgpucbf/xbgpu/engine.py:404]: Similarly, see here:
https://github.com/ska-sa/katgpucbf/blob/3b6cf9039403f5479e442b1d71f244f9820d4e1f/src/katgpucbf/xbgpu/engine.py#L332

**Review by bmerry — CHANGES_REQUESTED** [review]

File: src/katgpucbf/xbgpu/engine.py [hunk 3]
```diff
@@ -377,10 +377,37 @@ def _populate_sensors(self) -> None:
...
             )
             # Dynamic sensors
+            sensors.add(
+                aiokatcp.Sensor(
+                    str,
+                    f"{output.name}.delay",
+                    "The delay settings of the inputs for this beam: (loadmcnt "
+                    "<ADC sample count when model was loaded>, delay <in seconds>, "
+                    "phase <radians>, ...)",
+                    initial_status=aiokatcp.Sensor.Status.NOMINAL,
```
[the remark this answers is not in the document]
Comment (amishatishpatel) [reply src/katgpucbf/xbgpu/engine.py:387]: You know, I had that exact string in here, decided to remove it to get explicit feedback on the formatting 😅

**Review by bmerry — CHANGES_REQUESTED** [review]

File: src/katgpucbf/xbgpu/engine.py [hunk 4]
```diff
@@ -377,13 +377,16 @@ def _populate_sensors(self) -> None:
                 )
             )
             # Dynamic sensors
+            # Using the zeroth values for defaults as they're all the same at the start
+            default_delays_str = ", ".join(str(value) for value in self._delays[0].flatten())
```
[the remark this answers is not in the document]
Comment (amishatishpatel) [reply src/katgpucbf/xbgpu/engine.py:381]: Fair point! Change incoming.

File: src/katgpucbf/xbgpu/engine.py [hunk 5]
```diff
@@ -404,7 +403,8 @@ def _populate_sensors(self) -> None:
                     str,
                     f"{output.name}.weight",
                     "The summing weights applied to all the inputs of this beam",
-                    default=str(list(self._weights[0])),
+                    # Cast to list first to add comma delimeter
```
Comment (bmerry) [inline src/katgpucbf/xbgpu/engine.py:406]:
```suggestion
                    # Cast to list first to add comma delimiter
```

File: src/katgpucbf/xbgpu/engine.py [hunk 6]
```diff
@@ -377,10 +377,41 @@ def _populate_sensors(self) -> None:
                 )
             )
             # Dynamic sensors
+            default_delays_str = ", ".join(str(value) for value in self._delays[i].flatten())
+            sensors.add(
+                aiokatcp.Sensor(
+                    str,
+                    f"{output.name}.delay",
+                    "The delay settings of the inputs for this beam. Each input has "
+                    "a delay [s] and phase [rad]: (loadmcnt, delay0, phase0, delay1,"
```
Comment (bmerry) [inline src/katgpucbf/xbgpu/engine.py:386]:
```suggestion
                    "a delay [s] and phase [rad]: (loadmcnt, delay0, phase0, delay1, "
```

**Review by bmerry — CHANGES_REQUESTED** [review]
Just a couple of typos - feel free to merge after fixing them.

**Pull request merged**